python/pyarrow/tests/test_dataset.py

```diff
@@ -3095,10 +3104,24 @@ def test_write_dataset_use_threads(tempdir):
         pa.schema([("part", pa.string())]), flavor="hive")
 
     target1 = tempdir / 'partitioned1'
+    paths_written = []
+
+    def file_visitor(written_file):
+        paths_written.append(written_file.path)
+
     ds.write_dataset(
         dataset, target1, format="feather", partitioning=partitioning,
-        use_threads=True
+        use_threads=True, file_visitor=file_visitor
     )
+    expected_paths = [
+        target1 / 'part=a' / 'part-0.feather',
+        target1 / 'part=a' / 'part-1.feather',
+        target1 / 'part=b' / 'part-0.feather',
+        target1 / 'part=b' / 'part-1.feather'
+    ]
+    for path in paths_written:
+        assert pathlib.Path(path) in expected_paths
```

Comment:
This technically only requires that paths_written is a subset of expected_paths
```suggestion
    assert set(map(pathlib.Path, paths_written)) == {
        target1 / 'part=a' / 'part-0.feather',
        target1 / 'part=a' / 'part-1.feather',
        target1 / 'part=b' / 'part-0.feather',
        target1 / 'part=b' / 'part-1.feather',
    }
```

Reply:
That was intentional.  Only two paths are written but I don't know if part-0.feather goes in part=a or part=b (since it is multithreaded).  I refined the check to be a little clearer and added a comment.